pkg/cache/v3/snapshot.go

```diff
@@ -87,7 +87,10 @@ func NewResourcesWithTtl(version string, items []types.ResourceWithTtl) Resource
 type Snapshot struct {
 	Resources [types.UnknownType]Resources
 
-	// VersionMap holds the current hash map of all resources in the snapshot. This is only used in delta xDS
+	// VersionMap holds the current hash map of all resources in the snapshot.
+	// This field should remain nil until it is used, at which point should be
+	// instantiated by setting a snapshot while, or creating delta xDS resource watches.
```

Comment:
it would be instantiated by calling ComputeVersionMap or whatever it was called right?

Reply:
Yes that is correct. That `ConstructVersionMap` function gets called when creating watches or setting a snapshot thought it's probably better to say directly what instantiates that version map instead.